### PR TITLE
[opt](cast) Optimize the conversion of numeric types to strings.

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

before
```
mysql [(none)]>select sum(length(cast(number as string) ))   from numbers("number" = "100000000");
+-----------------------------------+
| sum(length(cast(number as TEXT))) |
+-----------------------------------+
|                         788888890 |
+-----------------------------------+
1 row in set (7.24 sec)
```

now
```
mysql [(none)]>select sum(length(cast(number as string) ))   from numbers("number" = "100000000");
+-----------------------------------+
| sum(length(cast(number as TEXT))) |
+-----------------------------------+
|                         788888890 |
+-----------------------------------+
1 row in set (1.40 sec)

mysql [test]>select sum(length(cast(123 as string) ))   from numbers("number" = "100000000");
+--------------------------------+
| sum(length(cast(123 as TEXT))) |
+--------------------------------+
|                      300000000 |
+--------------------------------+
1 row in set (0.09 sec)
```
More batch functions will be implemented in the future.

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Review by zhiqiang-hhhh — COMMENTED** [review]
Need ut

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run performance

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

File: be/src/vec/data_types/data_type_ipv4.cpp [hunk 1]
```diff
@@ -32,6 +32,16 @@ bool DataTypeIPv4::equals(const IDataType& rhs) const {
     return typeid(rhs) == typeid(*this);
 }
 
+size_t DataTypeIPv4::number_length() const {
+    //255.255.255.256
```
Comment (HappenLee) [inline be/src/vec/data_types/data_type_ipv4.cpp:36]: no 256

**Comment by Mryange** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Review by jacktengg — APPROVED** [review]

**Review by Gabriel39 — APPROVED** [review]

**Pull request merged**